Chessington.GameEngine/Pieces/Pawn.cs

```diff
@@ -43,40 +45,60 @@ public override IEnumerable<Square> GetAvailableMoves(Board board)
                 availableLocations.Add(twoSquaresAhead);
             }
 
-            // Pawns can take pieces diagonally
-            var diagonalLeft = GetDiagonalMove(board, location, HoriztonalDirection.Left);
-            if (diagonalLeft != null) availableLocations.Add(diagonalLeft.Value);
-
-
-            var diagonalRight = GetDiagonalMove(board, location, HoriztonalDirection.Right);
-            if (diagonalRight != null) availableLocations.Add(diagonalRight.Value);
-
-            return availableLocations;
+            // Pawns can take pieces diagonally, and en passant
+            return availableLocations
+                .Concat(GetDiagonalMoves(board, location, -1))
+                .Concat(GetDiagonalMoves(board, location, 1));
         }
 
-        private Square? GetDiagonalMove(Board board, Square location, HoriztonalDirection horizontalDirection)
+        private IEnumerable<Square> GetDiagonalMoves(Board board, Square location, int horizontalDirection)
```

Comment:
Did you intentionally change this back to `int` instead of the new enum? 🙂

Reply:
Nope, I think I did a dodgy rebase here. Sorry about that. Fixed in ba72ab6